builtin/providers/aws/resource_aws_db_subnet_group.go

```diff
@@ -166,8 +165,9 @@ func resourceAwsDbSubnetGroupUpdate(d *schema.ResourceData, meta interface{}) er
 		}
 
 		_, err := conn.ModifyDBSubnetGroup(&rds.ModifyDBSubnetGroupInput{
-			DBSubnetGroupName: aws.String(d.Id()),
-			SubnetIds:         sIds,
+			DBSubnetGroupName:        aws.String(d.Id()),
+			DBSubnetGroupDescription: aws.String(d.Get("description").(string)),
```

Comment:
I am writing a test for this right now based on a small tweak to how your code works. I will notify you in the commit so you can see the way we tend to handle this already - the work you have done is working but it will need to be changed pending any further Update params being required